rq/queue.py

```diff
@@ -106,6 +106,9 @@ def is_empty(self):
         return self.count == 0
 
     def fetch_job(self, job_id):
+        if job_id not in self.job_ids:
```

Comment:
In addition to adding an extra Redis call, this line will be really expensive to run on a long queue because it will fetch all `job_ids` in this queue and checks whether `job_id` is in that list.

A more performant approach would be to modify Line 113 to:
```python
try:
    job = self.job_class.fetch(job_id, connection=self.connection)
    if job.queue.name != self.name: # Something like this
        return
    return job
except NoSuchJobError:
```

Could you please update the PR?

This change is not backwards compatible so we'll have to note this in the changelog along with https://github.com/nvie/rq/pull/743#issuecomment-241607859